src/jquery.uls.lcd.js

```diff
@@ -59,13 +59,86 @@
 		this.$cachedQuicklist = null;
 		this.groupByRegionOverride = null;
 
+		// The index of the language option that is currently visited using arrow key navigation
+		// Can take values in the [0, languageOptionListItemsLength - 1] range for top to bottom navigation,
+		// or in the [-1, -languageOptionListItemsLength + 1] range for bottom to top navigation.
+		this.navigationIndex = null
+
 		this.render();
 		this.listen();
 	}
 
 	LanguageCategoryDisplay.prototype = {
 		constructor: LanguageCategoryDisplay,
 
+		/**
+		 * Returns a jQuery object containing a collection of all the
+		 * language option <li> elements
+		 * @return {jQuery}
+		 */
+		getLanguageOptionListItems() {
+			return this.$element.find( 'li[data-code]' );
+		},
+
+		/**
+		 * Increases the keyboard navigation index by one and applies a specific
+		 * class to the n-th language option <li> element (where n = navigation index)
+		 * Currently used as event handler for the arrow down 'keydown' event, inside
+		 * LanguageFilter.
+		 */
+		navigateDown: function () {
```

Comment:
The up / down arrow should take the currently selected option into consideration even if that selection was made using the mouse click. Would the current implementation do option?